### PR TITLE
Copy method/field descriptor on click

### PR DESCRIPTION
**Pull request by YanisBft**
Not revolutionary, but can be very useful when doing unpick definitions manually

### CODE REVIEW

**Review by liach — APPROVED** [review]
Reminds me of the scrambles when I tried to add copy on click mapping info to yarn javadocs

**Comment by liach** [comment]
Imo if we can show the original names of the elements than only the mapped ones when an element is selected, it would be even more helpful

**Comment by YanisBft** [comment]
> Imo if we can show the original names of the elements than only the mapped ones when an element is selected, it would be even more helpful

Out of scope, but that would be nice indeed. Maybe a toggle button to display obf/deobf? Something like show/hide password on website forms

**Comment by liach** [comment]
> 
> 
> > Imo if we can show the original names of the elements than only the mapped ones when an element is selected, it would be even more helpful
> 
> Out of scope, but that would be nice indeed. Maybe a toggle button to display obf/deobf? Something like show/hide password on website forms

Sure. Just make sure things don't bug out when you try to rename stuff while displaying obf

File: enigma-swing/src/main/java/cuchaz/enigma/gui/util/GuiUtil.java [hunk 1]
```diff
@@ -53,6 +54,24 @@ public static JLabel unboldLabel(JLabel label) {
...
     }
 
+    /**
+     * Puts the provided {@code text} in the system clipboard.
+     */
+    public static void copyToClipboard(String text) {
+        Toolkit.getDefaultToolkit().getSystemClipboard().setContents(new StringSelection(text), null);
+    }
+
+    public static void showPopup(JComponent component, String text, int x, int y) {
```
Comment (2xsaiko) [inline enigma-swing/src/main/java/cuchaz/enigma/gui/util/GuiUtil.java:64]: Personally I'd put that message in the status bar but this works :P

**Review by 2xsaiko — APPROVED** [review]

File: enigma-swing/src/main/java/cuchaz/enigma/gui/util/GuiUtil.java (hunk 1 above)
Comment (YanisBft) [reply enigma-swing/src/main/java/cuchaz/enigma/gui/util/GuiUtil.java:64]: I think this is too far away from the copied text, I don't think people would notice it

**Pull request merged**